---
string[] tote_contents
string[] bin_contents
